README.md

```diff
@@ -84,6 +85,21 @@ steps:
       github_token: ${{ github.token }}
 ```
 
+### Buf username and Buf API token
```

Comment:
```suggestion
### Authenticating with the Buf Schema Registry
```

Reply:
maybe also worth putting a link to the docs site "for more information" here unless its already mentioned

Reply:
This part is quite confusing since it mentions parameters and other configurations in the configurations in the same section. I think it might be a better idea to separate the parameters and other configurations.

Reply:
I also see another problem, we speak about authenticating with the BSR using 2 different methods: on with a username and api token, the other with just a token. Is there a way we might be able to distinguish between the two usecases?
1. The .netrc file allows users to quickly login to the BSR system-wide without manually having to reenter the credentials for each tool(includes private go modules).  
2. The token env var authenticates just `Buf`, not the whole system.

Reply:
I made the changes:
1. I mentioned the access to private remote packages in the section of supplying username and buf api token
2. I also mentioned in the Buf Token part that specifying BUF_TOKEN as env var does does gain the access to private remote packages. 

I will bring this up briefly in the doc site.